### PR TITLE
feat: new sidebar component

### PR DESCRIPTION
**Pull request by joyce-shi**
## Notion ticket link

[Updated Sidebar Component](https://www.notion.so/uwblueprintexecs/Updated-Sidebar-Component-b1e3805085ca418c803318e4c7f3ff6b)

## Implementation description
Implementing the new navigation bar requested by design.

## Steps to test
1. Test navigation bar behaviour.
2. Test children page navigation bar behaviour.

See the design linked [here](https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=1538%3A79443&t=8UlpYW8GI1IfD3HR-4).

https://user-images.githubusercontent.com/41273486/210194602-779babd8-a42f-49d7-ae73-5ac954896c53.mov

### CODE REVIEW

File: frontend/src/components/pages/ComponentLibrary.tsx [hunk 1]
```diff
@@ -187,12 +186,11 @@ const pages: Page[] = [
...
-    icon: BookIcon,
   },
-  { title: "Teacher Page", url: Routes.TEACHER_PAGE, icon: BookIcon },
-  { title: "Admin Page", url: Routes.ADMIN_PAGE, icon: BookIcon },
-  { title: "Library", url: Routes.COMPONENT_LIBRARY, icon: BookIcon },
-  { title: "Assessments", url: "/", icon: BookIcon },
+  { title: "Teacher Page", url: Routes.TEACHER_PAGE },
+  { title: "Admin Page", url: Routes.ADMIN_PAGE },
+  { title: "Library", url: Routes.COMPONENT_LIBRARY },
+  { title: "Assessments", url: "/" },
```
Comment (carissa-tang) [inline frontend/src/components/pages/ComponentLibrary.tsx:193]:
```suggestion
  { title: "Assessments", url: "/assessments" },
```

File: frontend/src/components/pages/AdminPage.tsx [hunk 2]
```diff
@@ -20,21 +19,17 @@ import {
...
-} from "../../assets/icons";
+import { AlertIcon, SearchOutlineIcon } from "../../assets/icons";
 import GET_USERS_BY_ROLE from "../../APIClients/queries/UserQueries";
 import SortTablePopover from "../common/SortTablePopover";
 
 const pages: Page[] = [
-  { title: "Assessments", url: "/", icon: SettingsOutlineIcon },
-  { title: "Database", url: "/", icon: SettingsOutlineIcon },
+  { title: "Assessments", url: "/assessments" },
+  { title: "Database", url: "/admin-dashboard" },
```
Comment (carissa-tang) [inline frontend/src/components/pages/AdminPage.tsx]:
```suggestion
  { title: "Database", url: Routes.ADMIN_PAGE },
```

File: frontend/src/components/pages/AdminPage.tsx [hunk 3]
```diff
@@ -20,21 +19,17 @@ import {
...
-  SearchOutlineIcon,
-} from "../../assets/icons";
+import { AlertIcon, SearchOutlineIcon } from "../../assets/icons";
 import GET_USERS_BY_ROLE from "../../APIClients/queries/UserQueries";
 import SortTablePopover from "../common/SortTablePopover";
 
 const pages: Page[] = [
-  { title: "Assessments", url: "/", icon: SettingsOutlineIcon },
-  { title: "Database", url: "/", icon: SettingsOutlineIcon },
+  { title: "Assessments", url: "/assessments" },
```
Comment (carissa-tang) [inline frontend/src/components/pages/AdminPage.tsx]: can we add a route  for `ASSESSMENTS`?

File: frontend/src/components/pages/AdminPage.tsx [hunk 4]
```diff
@@ -20,21 +19,17 @@ import {
...
-  SettingsOutlineIcon,
-  AlertIcon,
-  SearchOutlineIcon,
-} from "../../assets/icons";
+import { AlertIcon, SearchOutlineIcon } from "../../assets/icons";
 import GET_USERS_BY_ROLE from "../../APIClients/queries/UserQueries";
 import SortTablePopover from "../common/SortTablePopover";
 
 const pages: Page[] = [
-  { title: "Assessments", url: "/", icon: SettingsOutlineIcon },
```
Comment (carissa-tang) [inline frontend/src/components/pages/AdminPage.tsx:36]: Should there also be a `Dashboard` option that points to the admin page itself?

Comment (joyce-shi) [reply frontend/src/components/pages/AdminPage.tsx:36]: I think `Database` refers to the dashboard? If that's what you mean?

File: frontend/src/components/pages/AdminPage.tsx (hunk 3 above)
Comment (joyce-shi) [reply frontend/src/components/pages/AdminPage.tsx]: good catch! adding.

File: frontend/src/components/pages/AdminPage.tsx (hunk 4 above)
Comment (carissa-tang) [reply frontend/src/components/pages/AdminPage.tsx:36]: ohh i think i got confused cause i see these three options in Figma 
image

Comment (joyce-shi) [reply frontend/src/components/pages/AdminPage.tsx:36]: oh hmmm... i think there was some talk of adding an admin landing dashboard? but im not sure if we're still doing that 🤨, we can talk more  bout this during our next sync :P

**Review by carissa-tang — APPROVED** [review]
lgtm!

**Pull request merged**